sys/include/net/gnrc/netreg.h

```diff
@@ -334,17 +334,20 @@ void gnrc_netreg_unregister(gnrc_nettype_t type, gnrc_netreg_entry_t *entry);
 gnrc_netreg_entry_t *gnrc_netreg_lookup(gnrc_nettype_t type, uint32_t demux_ctx);
 
 /**
- * @brief   Returns number of entries with the same gnrc_netreg_entry_t::type and
- *          gnrc_netreg_entry_t::demux_ctx.
+ * @brief   Searches the next entry in the registry that matches given
+ *          parameters, start lookup from beginning or given entry.
  *
+ * @param[in] from      A registry entry to lookup from or NULL to start fresh
  * @param[in] type      Type of the protocol.
  * @param[in] demux_ctx The demultiplexing context for the registered thread.
  *                      See gnrc_netreg_entry_t::demux_ctx.
  *
- * @return  Number of entries with the same gnrc_netreg_entry_t::type and
- *          gnrc_netreg_entry_t::demux_ctx as the given parameters.
+ * @return  The first entry fitting the given parameters on success
+ * @return  NULL if no entry can be found.
  */
-int gnrc_netreg_num(gnrc_nettype_t type, uint32_t demux_ctx);
+gnrc_netreg_entry_t *gnrc_netreg_lookup_next(gnrc_netreg_entry_t *from,
+                                             gnrc_nettype_t type,
+                                             uint32_t demux_ctx);
```

Comment:
This new function looks from its description awfully lot like `gnrc_netreg_getnext()`, only that you can change `type` and `demux_ctx` during iteration (?) which seems dangerous to me.

Reply:
mhm, making it private (as you suggested, below) would make it less dangerous, right? originally I started off with this function to replace all usages of `gnrc_netreg_lookup()`, `gnrc_netreg_getnext()`, and `gnrc_netreg_num`. However, in the process I discovered that the first two have their merits in providing often used short hands. 

So I tend to agree with making it private, but will also check how this impacts binary size, first.

Reply:
as stated in my other comments, the idea was to provide a (more) generic function to replace `gnrc_netreg_getnext` - at least that was my original plan.